### ISSUE TITLE
IE11 links in document preview iframe not opening

### ISSUE THREAD

**Issue by Guillaume-Developer** [sinequa/sba-angular#20]
I have an issue with document preview containing links inside its html (using an html template, the document preview was constructed with href links inside it). In Chrome, the links open without any problem, but in IE11, it doesn't open whatever I do. I added the "allow-popups" and all "allow-..." from the "defaultSandbox", but it still doesn't work.

![image](https://user-images.githubusercontent.com/65693851/105264358-ef8aa500-5bd3-11eb-8a7b-d5f33cc89335.png)

The link is a simple `<a >` tag inside the iframe: 

`<a style="left: 0px; top: 0px; width: 100%; height: 100%; position: absolute; cursor: pointer;" href="https://pubmed.ncbi.nlm.nih.gov/" target="_blank"></a>`

**Comment by ericleib** [comment]
Strange, I never faced that problem. Do you have the same behaviour with profiles ?

**Comment by Guillaume-Developer** [comment]
No, the links open without any issue when using the profiles.

**Comment by ericleib** [comment]
And does it only occur with links generated with HTML templates, or is it the case with any type of document ?

**Comment by Guillaume-Developer** [comment]
Same behavior occurs with other type of document (I tried this with an indexed website). The only thing that is not working is opening a link outside of the IFrame, but it can open inside it. I think this has to do with IE11 policy: 

> In IE 11, only links to the same domain work in IFRAMEs. Linking to domains other than the one the page loaded from is prohibited now.

--> https://stackoverflow.com/questions/31608674/base-target-blank-not-working-for-iframe-in-ie11

**Comment by ericleib** [comment]
Interesting, it seems it might have something to do with intranet/internet zones:

https://social.msdn.microsoft.com/Forums/en-US/b1c3cb0d-1947-49c1-b5df-9423e64cb0ce/links-that-open-in-new-browser-tabs-dont-work-on-sites-embedded-within-intranet-zone-sites-in-ie

**Comment by Guillaume-Developer** [comment]
I see... Any solutions that we might be able to apply to fix this? The fact that it was working fine with the profiles makes it look like a regression to the customer, so we can't really say "It's just IE policy, nothing to do about it." since they will ask why we don't do the same as what we did in profiles.

**Comment by ericleib** [comment]
I don't see a lot that we can do... Profiles were using frameset/frames instead of iframes, but I don't think we want to go back to that.
One thing that might work, but is not very clean: add an event listener to the document to detect clicks on links with target=_blank. If the user is using IE, then open these links manually on the Angular side (since we know IE refuses to open them).

**Comment by Guillaume-Developer** [comment]
Unfortunately it seems that the document listener for document click doesn't detect the clicks in an iframe. So it looks like this solution will not work either. 
Any other ideas?

**Comment by ericleib** [comment]
Yes, actually you need to add the listener to the iframe's document. Something like:

```
previewDocument.document.addEventListener("click", event => {
  ...
});
```
